File: Programmers/Sort_BiggestNumber.cpp
//프로그래머스 정렬 #2 가장 큰 수, 성공

#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool compare(const string &a, const string &b){
    return (a+b) > (b+a);
}
string solution(vector<int> numbers) {
    string answer = "";
    vector<string> strnumbers;

    for(int i=0; i<numbers.size(); i++){
        strnumbers.push_back(to_string(numbers[i]));
    }

    sort(strnumbers.begin(), strnumbers.end(), compare);

    if(strnumbers[0] == "0") return "0";

    for(int i=0; i<strnumbers.size(); i++){
        answer += strnumbers[i];
    }

    return answer;
}
int main(){

    string ans = solution({6,10, 2});

    cout<<ans<<'\n';
    return 0;
}

File: Programmers/Sort_BiggestNumber.swift

import Foundation

// 프로그래머스 가장 큰 수, 정렬, level 2

func cmp(_ str1: String, _ str2: String) -> Bool {
    if str1+str2 > str2+str1 {
        return true
    }
    return false
}

func solution(_ numbers:[Int]) -> String {
    var array = numbers.map{String($0)}
    array.sort(by: cmp)
    
    if array[0] == "0" {
        return "0"
    }
    
    let answer = array.reduce(""){$0+$1}
    
    return answer
}
